src/main/java/com/parasoft/report/transformer/XMLToSarif.java

```diff
@@ -118,4 +136,25 @@ private void convertXmlToSarif() {
         }
         Logger.info(MessageFormat.format("SARIF report has been created: {0}", this.outputSarifReport.getAbsolutePath()));
     }
+
+    private boolean isAbsolutePath(String path) {
+        Pattern pattern = Pattern.compile(ABSOLUTE_PATH_REGEX);
+        return pattern.matcher(path).matches();
+    }
+
+    private void avoidDuplicateProjectRootPaths(String[] paths) {
+        Set<String> uniquePaths = new HashSet<>();
+        for (String path : paths) {
+            path = path.endsWith("/") ? path : path + "/";
+            if (uniquePaths.contains(path)) {
+                throw new IllegalArgumentException(MessageFormat.format("Duplicate project root path found: {0}", path));
+            }
+            for (String uniquePath : uniquePaths) {
+                if (path.startsWith(uniquePath) || uniquePath.startsWith(path)) {
+                    throw new IllegalArgumentException(MessageFormat.format("Project path contains or is contained by another path: {0} and {1}", path, uniquePath));
```

Comment:
How about:
`Project path conflict: Path '{0}' contains or is contained by '{1}', which is not supported.`